andela/elven-ah
# #159204726 Send notifications on article and user activities

#### What does this PR do?
Add ability for users to notification when there is an activity on an article they have interacted with, when there is an activity on their article and when another user follows them.
#### Description of Task to be completed?
- Create Channel, Subscription and Notification
- Create Channel, Subscription and Notification migrations
- create Notification controller
- add methods for subscribing to and sending notification
- add create a user channel on signup
- add delete an article channel when the article is deleted
- add unsubscribe method to unsubscribe from a notification
- add public folder with a sample implementation of notification on the frontend

#### How should this be manually tested?
- Clone the repo: Run `git clone https://github.com/andela/elven-ah.git` in the folder where you want the repo to be saved.
- Navigate into the newly created folder and install the dependencies: `cd elven-ah && npm install`
- Copy .env.example to .env: `cp .env.example .env`
- Provision a PostgreSQL database and add the credentials to the .env file. 
- Start the server using `npm start`
- create an account  and create an article or seed some articles
- Open http://localhost:3000 in a browser and log in with any of the seeded users
- Then using postman comment/rate articles with the token of the user logged in above
- comment on the same article with other users' tokens and notice the alerts on the frontend
#### What are the relevant pivotal tracker stories? 
[#159204726](https://www.pivotaltracker.com/story/show/159204726)
